### ISSUE TITLE
AttackEngine crashes with a lot of garbage queued

### ISSUE THREAD

**Issue by Endaris** [panel-attack/panel-game#374]
```
Error

Stack Trace:     engine/telegraph.lua: in function 'chainingEnded'
    AttackEngine.lua:123: in function 'run'
    SimulatedOpponent.lua:30: in function 'run'
    match.lua:230: in function 'run'
    mainloop.lua:387: in function 'f'
    mainloop.lua:115: in function 'variable_step'
    mainloop.lua:384: in function 'func'
Username: WatchgangAmadeus
Theme: Panel Attack Classic
Error Message: main.lua:148: engine/telegraph.lua:165: attempt to index local 'chain' (a nil value)
Engine Version: 047
Build Version: Unknown
Operating System: OS: OS X
Love Version: 11.4.0
UTC Time: 2023-11-01 00:32:02
vs Match Info: 
  Stage: ppl_friends2
  Stacks: 
    P1: 
      Player Number: 1
      Character: puyo_suketoudarajpn_shadow
      Panels: panelhd_basic_mizunoketsuban
      Rollback Count: 0
      Rollback Frames Saved: 0
```

Reported to happen after 4-10min on training files that send high gpm (30+).
A guess is that the crash does only happen once the limit for queued garbage via AttackEngine is being reached (currently 72 pieces? lines? of garbage).

**Comment by Endaris** [comment]
The issue here is that gradual build-up of chains is being segmented into separate parts of the attack pattern.
Once the recipient's incoming garbage queue is above the limit, the attack engine starts *skipping* patterns because it prioritizes being faithful to its clock rather than its garbage sent.
That in particular can lead to the recipient's incoming garbage queue to slip just below the limit in a moment when the next attack pattern of the attack engine is an "end chain" entry.
But the chain it was supposed to end got entirely skipped so there is no chain and we crash.

This is also an indicator for the attack engine skipping entries and producing inconsistent garbage that depends largely on when the recipient is letting garbage drop rather than the attack patterns the engine has loaded.
Ideally a solution would also find a way for garbage to be consistent.

For the time being I have set all attack pattern dumps from human player stacks to ignore the queue limit so that realistically achieveable garbage output cannot reproduce the problem and prevented the crash by breaking early which is a bandaid but not a solution.
The queue limit mainly exists to accomodate training modes like combo storm because without a queue limit it would just flood the memory with garbage and eventually crash. In training modes this also should not crash because chains are being submitted in the form of combos there.

Realistically we may want to tackle this in conjunction with https://github.com/panel-attack/panel-game/issues/325 in some way.

**Comment by Endaris** [comment]
This should have been fixed between 561e5ddd0939d8b2ad985351a3a5aa7c7d5298f2 and c46c386fb1c02a47c3efbe569670b1e33d059182 

Note that the attack engine still starts behaving somewhat haphazardly after hitting the queue limit but it's mostly not a practical issue I think.
